src/script_opt/UsageAnalyzer.cc

```diff
@@ -122,8 +122,12 @@ void UsageAnalyzer::FindSeeds(IDSet& seeds) const
 
 		// If the global is exported, or has global scope, we assume
 		// it's meant to be used, even if the current scripts don't
-		// use it.
-		if ( id->IsExport() || id->ModuleName() == "GLOBAL" )
+		// use it.  Similarly, lambdas are either attributes, in
+		// which case we should presume they're used, or appear inside
+		// of functions, so if not used there will be a broader
+		// complaint about the encompassing function.
+		if ( id->IsExport() || id->ModuleName() == "GLOBAL" ||
+		     std::string(id->Name()).find("lambda_<") != std::string::npos )
```

Comment:
Hey - I think this works for ones dinged in #3122 and probably suffices to avoid false positives, think that's good to merge unless you see an obvious fix for below.

Poking at the fix a bit harder, the following would be cases that aren't covered, but can see them as not idiomatic and probably can ignore. The &default usage on the record field is a bit surprising though, because that should be found, no?

```
module MyRadius;

function default_id(c: count): string {
        return fmt("unknown-%d", c);
}

const other_ids: table[count] of string = {
        [1] = "One",
} &default=default_id;

function default_id2(): string {
        return "";
}

const more_ids: table[count] of string = {
        [1] = "One",
} &default=default_id2() + "";


function gen_id(): string {
        return cat(rand(10000));
}

type R: record {
        id: string &default=gen_id();
};

event zeek_init() {
        print R();
}

warning in ./x.zeek, line 3: non-exported function does not have any callers (MyRadius::default_id)
warning in ./x.zeek, line 11: non-exported function does not have any callers (MyRadius::default_id2)
warning in ./x.zeek, line 20: non-exported function does not have any callers (MyRadius::gen_id)
[id=3890]
```

Reply:
Dang, I clearly wasn't thinking about the problem in detail.  Let me mull over how hard it is to fix things correctly so it takes care of those, too.

Reply:
Vern / me chatted and have an idea for a more general fix, will update the PR tomorrow and see if that fits in well.